javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinter.java

```diff
@@ -189,7 +230,7 @@ public void concretePropertyChange(Node observedNode, ObservableProperty propert
                 matchingTokens = nodeText.getElements().stream()
                     .filter(e -> e.isToken(SINGLE_LINE_COMMENT))
                     .map(e -> (TokenTextElement) e)
-                    .filter(t -> t.getText().equals("//" + oldValue.getContent() + System.lineSeparator()))
```

Comment:
Ah yes, this could never have gone right because you never know what EOL's are in the input.

Reply:
Yeah, that was one part of the problem. The other was that the comment is in a TokenTextElement with a Windows file, but in a ChildTextElement with a Unix file

Reply:
It sounds like something else is wrong there. There are several EOL tokens, and it seems there is a hard comparison with one of them somewhere in the code :-/